src/ape/api/projects.py

```diff
@@ -113,10 +113,42 @@ def _create_manifest(
         if version:
             manifest.version = version
 
+        manifest.compilers = cls._create_compilers_list(source_paths, contract_types)
         manifest.sources = cls._create_source_dict(source_paths, contracts_path)
         manifest.contract_types = contract_types
+
         return manifest
 
+    @classmethod
+    def _create_compilers_list(
+        cls,
+        source_paths: List[Path],
+        contract_types: Dict[str, ContractType],
+    ):
+        compiler_list: List[Compiler] = []
+        for key, compiler in cls.compiler_manager.registered_compilers.items():
+            filtered_contract_types = [
+                str(x)
+                for x in contract_types.keys()
+                if key.strip(".") in contract_types[x].source_id.split(".")[-1]
+            ]
+            compiler_list.append(
+                Compiler(
+                    name=compiler.name,
+                    # TODO: Figure out if this should be str
+                    version=str(compiler.get_versions(source_paths)),
```

Comment:
We should be looping through the versions used in the project and attaching individual instances of `Compiler` per version (which are str then instead of a stringified set like here)

I fear I misled you! @NotPeopling2day suggested this approach but I think it would be inaccurate to include versions of compilers not used in the active project.

Reply:
ah nevermind, we use the sources are the key so this is good.

but we just need to loop over the versions then

Reply:
NOTE: Take care to only include contract types that used that specific version though.. I am not sure if are we able to do this using `get_versions()` because it only returns a `set` whereas the method in `ape-solidity` `get_version_map()` returns a dictionary that maps version to contract types.

We may just want to add `get_version_map` to the API...

Reply:
yes you're right! 
```
[Compiler(name='solidity', version="{'0.8.15', '0.6.12'}", settings={}, contractTypes=['SafeMath', 'Token', 'Payee', 'GenerativeNFT']),
 Compiler(name='vyper', version="{'0.2.16'}", settings={}, contractTypes=['VyperToken']),
 Compiler(name='ethpm', version='set()', settings={}, contractTypes=[])]
 ```
 this object doesn't assign the correct versions when there's more than one version of solidity file being compiled simultaneously! 😅